CMakeLists.txt

```diff
@@ -311,10 +311,125 @@ add_subdirectory(pkgconfig)
 # to define the test executables.
 #
 if(ENABLE_TESTING OR ENABLE_PROGRAMS)
+    add_custom_command(
```

Comment:
All this stuff should be in `tests/CMakeLists.txt` rather than the toplevel file.